pytorch_lightning/utilities/device_dtype_mixin.py

```diff
@@ -121,11 +121,16 @@ def cuda(self, device: Optional[Union[torch.device, int]] = None) -> Module:
         Returns:
             Module: self
         """
-        property_device = device if isinstance(device, torch.device) else torch.device('cuda', index=device)
```

Comment:
why was there a need to change this?
the last else you added is already covered in the previous one because `device=None` leads to `index=None`.

Reply:
From some reasons, *mypy* expects `index` should be of a type `int`, not `Optional[int]`. I'm gonna try some workarounds to make this nicer.

Reply:
I think this is a bug in the torch device typing definition, as the index is clearly optional. Maybe we should revert this change and ignore it.

Reply:
Yes, it definitely looks like `device` of a type `Optional[int]` should be okay for the `index` argument. I reverted that change and added `# type: ignore` so that `mypy` checks will run smoothly for now.